test/Microsoft.Framework.DependencyInjection.Tests/ServiceProviderExtensionsTest.cs

```diff
@@ -41,31 +42,87 @@ public void NonGeneric_GetRequiredServices_ReturnsAllServices()
 
             // Act
             var services = serviceProvider.GetRequiredServices<IFoo>();
 
             // Assert
-            Assert.Contains(services, item => item is Foo1);
+            var item = Assert.Single(services);
+            Assert.IsType<Foo1>(item);
             Assert.Equal(1, services.Count());
         }
 
         [Fact]
-        public void ThrowsWhenNoServicesAreRegistered()
+        public void NonGeneric_GetRequiredServices_Returns_SingleService()
         {
             // Arrange
-            var serviceProvider = CreateFooServiceProvider(0);
+            var serviceProvider = CreateTestServiceProvider(1);
+
+            // Act
+            var services = serviceProvider.GetRequiredServices(typeof(IFoo));
+
+            // Assert
+            var item = Assert.Single(services);
+            Assert.IsType<Foo1>(item);
+            Assert.Equal(1, services.Count());
+        }
+
+        [Fact]
+        public void GetRequiredServices_Returns_CorrectTypes()
+        {
+            // Arrange
+            var serviceProvider = CreateTestServiceProvider(4);
+
+            // Act
+            var services = serviceProvider.GetRequiredServices(typeof(IAnotherFoo));
+
+            // Assert
+            Assert.True(services.All(s => s is IAnotherFoo));
+            Assert.Equal(2, services.Count());
+        }
+
+        [Fact]
+        public void Throws_WhenNo_Services_AreRegistered()
+        {
+            // Arrange
+            var serviceProvider = CreateTestServiceProvider(0);
 
             // Act + Assert
-            Assert.Throws<InvalidOperationException>(() => serviceProvider.GetRequiredServices<IFoo>());
+            ExceptionAssert.Throws<InvalidOperationException>(() => serviceProvider.GetRequiredServices<IFoo>(),
+                string.Format("No service for type '{0}' has been registered.", typeof(IFoo)));
         }
 
         [Fact]
-        public void ThrowsWhenNoServicesAreAvailable()
+        public void Throws_WhenNo_Services_AreRegistered_NonGeneric()
+        {
+            // Arrange
+            var serviceProvider = CreateTestServiceProvider(0);
+
+            // Act + Assert
+            ExceptionAssert.Throws<InvalidOperationException>(() => serviceProvider.GetRequiredServices(typeof(IFoo)),
+                string.Format("No service for type '{0}' has been registered.", typeof(IFoo)));
+        }
+
+        [Fact]
+        public void Throws_WhenNo_Services_AreAvailable()
+        {
+            // Arrange
+            var serviceCollection = new ServiceCollection();
+            serviceCollection.AddInstance<IEnumerable<IFoo>>(new List<IFoo>());
+
+            var serviceProvider = serviceCollection.BuildServiceProvider();
+
+            // Act + Assert
+            ExceptionAssert.Throws<InvalidOperationException>(() => serviceProvider.GetRequiredServices<IFoo>(),
+                string.Format("No service for type '{0}' has been registered.", typeof(IFoo)));
```

Comment:
This is fine as is, but you could write this as `$"No service for type '{typeof(IFoo)}' has been registered."`